plugins/module_utils/network/sonic/config/lldp_interfaces/lldp_interfaces.py

```diff
@@ -0,0 +1,374 @@
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+        diff = get_diff(want, have)
+        if state == 'deleted':
+            commands, requests = self._state_deleted(want, have, diff)
+        elif state == 'merged':
+            commands, requests = self._state_merged(diff)
+        elif state == 'replaced':
+            commands, requests = self._state_replaced(diff, want, have)
+        elif state == 'overridden':
+            commands, requests = self._state_overridden(want, have, diff)
+        return commands, requests
+
+    def _state_merged(self, diff):
+        """ The command generator when state is merged
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        commands = diff
+        requests = []
+        for command in commands:
+            requests.extend(self.get_modify_specific_lldp_interfaces_param_requests(command))
+        if commands and len(requests) > 0:
+            commands = update_states(commands, 'merged')
+        else:
+            commands = []
+        return commands, requests
+
+    def _state_deleted(self, want, have, diff):
+        """ The command generator when state is deleted
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+        requests = []
+        if not want:
+            commands = have
+            requests.extend(self.get_delete_lldp_interfaces_complete_requests(commands))
+        else:
+            commands = get_diff(want, diff)
+            for command in commands:
+                requests.extend(self.get_delete_specific_lldp_interfaces_param_requests(command, have))
+
+        if len(requests) == 0:
+            commands = []
+
+        if commands:
+            commands = update_states(commands, "deleted")
+
+        return commands, requests
+
+    def _state_replaced(self, diff, want, have):
+        """ The command generator when state is replaced
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        if not diff:
+            return commands, requests
+        commands = want
+        for command in commands:
+            # Get interface name and make all configs to default in that interface
+            command = {"name": command["name"]}
+            requests.extend(self.get_delete_specific_lldp_interfaces_param_requests(command, have))
```

Comment:
the "commands" should be updated.

Reply:
The rewritten version of this block looks fine.

Resolved.